import { LoginPage } from '../../support/pages/login.page'

it('shows a login error', () => {
  cy.visit('/')
  cy.get('[data-test="username"]').type('locked_out_user')
  cy.get('[data-test="password"]').type('secret_sauce')
  cy.log('**there are no errors**')
  cy.contains('[data-test=error]', 'locked out').should('not.exist')
  cy.get('[data-test="username"]').should('not.have.class', 'error')
  cy.get('[data-test="password"]').should('not.have.class', 'error')
  cy.get('[data-test="login-button"]').click()
  cy.log('**shows errors**')
  cy.get('[data-test="username"]').should('have.class', 'error')
  cy.get('[data-test="password"]').should('have.class', 'error')
  cy.location('pathname').should('equal', '/')
  cy.contains('[data-test=error]', 'locked out')
    .should('be.visible')
    .wait(1000)
    .find('button.error-button')
    .click()
  cy.log('**errors go away**')
  cy.contains('[data-test=error]', 'locked out').should('not.exist')
  cy.get('[data-test="username"]')
    .should('not.have.class', 'error')
    .and('have.value', 'locked_out_user')
  cy.get('[data-test="password"]')
    .should('not.have.class', 'error')
    .and('have.value', 'secret_sauce')
})

it('shows a login error refactored', () => {
  cy.visit('/')
  LoginPage.getUsername().type('locked_out_user')
  LoginPage.getPassword().type('secret_sauce')
  LoginPage.noErrors()
  cy.get('[data-test="login-button"]').click()
  cy.log('**shows errors**')
  LoginPage.getUsername().should('have.class', 'error')
  LoginPage.getPassword().should('have.class', 'error')
  cy.location('pathname').should('equal', '/')
  LoginPage.getError()
    .should('include.text', 'locked out')
    .and('be.visible')
    .wait(1000)
    .find('button.error-button')
    .click()
  LoginPage.noErrors()
  LoginPage.getUsername().should('have.value', 'locked_out_user')
  LoginPage.getPassword().should('have.value', 'secret_sauce')
})
